File: 백준/Bronze/2577. 숫자의 개수/숫자의 개수.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main{
    static int a,b,c,result,tmp;
    static int[] arr = new int[10];
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        a = Integer.parseInt(br.readLine());
        b = Integer.parseInt(br.readLine());
        c = Integer.parseInt(br.readLine());

        result = a * b * c;
        while(true){
            if(result/10 == 0 && result % 10 == 0) break;
            else {
                tmp = result % 10;
                arr[tmp]++;
                result /= 10;
            }
        }
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        for (int i = 0; i < arr.length; i++){
            bw.write(arr[i]+"\n");
        }
        bw.close();
}
}

File: 백준/Bronze/2577. 숫자의 개수/숫자의 개수.cc
#include<stdio.h>
#define SIZE 10

int main()
{
	int a, b, c, result,i;
	scanf("%d %d %d", &a, &b, &c);
	result = a * b * c;
	int arr[SIZE] = { 0, };
	int count[10] = { 0, };
	while (result > 0)
	{
		
		for (int i = 0; i < 10; i++)
		{
			if (result % 10 == i) count[i]++;
		}
		result /= 10;
	}
	for (int j = 0; j < 10; j++)
		printf("%d\n", count[j]);
	return 0;
}